src/qt/fontenum.cpp

```diff
@@ -7,13 +7,23 @@
 
 // For compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
-
 #include "wx/fontenum.h"
+#include <QFontDatabase>
 
-bool wxFontEnumerator::EnumerateFacenames( wxFontEncoding WXUNUSED(encoding),
-                bool WXUNUSED(fixedWidthOnly))
+bool wxFontEnumerator::EnumerateFacenames( wxFontEncoding WXUNUSED(encoding), bool fixedWidthOnly)
 {
-    return false;
+    QFontDatabase fontDatabase;
+    const QStringList allFonts = fontDatabase.families(QFontDatabase::Any);
+    for( QStringList::const_iterator i = allFonts.begin(); i != allFonts.end(); ++i)
+    {
+        const QString &fontFamily = *i;
+        if ( !fixedWidthOnly || fontDatabase.isFixedPitch(fontFamily) )
+        {
+            this->OnFacename(fontFamily.toStdString());
```

Comment:
- `std::string` is not always implicitly convertible to `wxString`; it should use the conversion helper;
- `OnFacename()` can return false, in which case enumeration should break, see the docs;
- `this->` is not used by default in wxW sources.

Reply:
What's the safe way to convert from std::string to wxString?

Reply:
> What's the safe way to convert from std::string to wxString?

You need to convert from `QString` to `wxString` (search for `wxQtConvertString`). You don't need `std::string` at all.

Reply:
Cool,  thank you.  I'll implement that change.